xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/chaining/AbstractChainingListener.java

```diff
@@ -55,14 +55,38 @@
      * @since 11.10.2
      * @since 12.0RC1
      */
-    private boolean beginListItemRetroCompatibility;
+    private final boolean beginListItemRetroCompatibility;
+
+    /**
+     * True if {@link #endListItem(Map)} should redirect to {@link #endListItem()} for retro compatibility.
+     * <p>
+     * {@link #endListItem(Map)} was added long after {@link #endListItem()} and various renderers/listeners
+     * override {@link #endListItem()} only and won't be called in post 10.0 versions of XWiki without this retro
+     * compatibility trick.
+     *
+     * @since 13.10.3
+     * @since 14.1RC1
+     */
+    private final boolean endListItemRetroCompatibility;
```

Comment:
I don't think it's accurate to have two different markers as begin and end calls are supposed to be in sync.

Reply:
So what should be the condition for enabling it? Should it only depend on `beginListItem`? Or should we check if one of the two misses the second method, or if both miss it? In particular in the light of https://jira.xwiki.org/browse/XRENDERING-646 this seems quite complex. In my opinion, we should not enable it too often as it is easy to add compatibility in a child class, but difficult to undo the compatibility added by `AbstractChainingListener` if you want to make use of the forwarding code it provides.

Reply:
IMO if any of begin or end with parameter is implemented then no retro compatibility (the point is to support implementation too old to know about the new list item calls).